UI for native projects in NewProject panel

rkzel opened the pull request:
As per design. Default view (right now GH connectivity is required on App level):

![Screenshot from 2020-01-22 14-45-33](https://user-images.githubusercontent.com/34452131/72899422-40985c80-3d26-11ea-8cdc-f77ba3817f9a.png)

When user start typing a title:

![Screenshot from 2020-01-23 20-00-06](https://user-images.githubusercontent.com/34452131/73017935-e3350600-3e20-11ea-97c8-a7904ed2912c.png)

And when GH is disconnected:

![Screenshot from 2020-01-23 19-59-40](https://user-images.githubusercontent.com/34452131/73017906-d87a7100-3e20-11ea-96fb-ef76d27f3ec7.png)

rkzel committed:
feat: added UI for native projects to NewProject panel

stellarmagnet reviewed (dismissed):
- The padding is not correct between the field labels. Please double check all of the padding between elements.
- For the "And when GH is disconnected" use case --> the blue box is too close to the field label
- "Feedback" should say "Description"
- Select from Github icon & text should be center-aligned with the button

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -113,110 +121,187 @@ const NewProject = () => {
-                      adornmentPosition="end"
-                      ref={searchRef}
-                      aria-label="Search"
+  const RepoQuery = () => (
```
Why define this inside `GithubRepoList` instead of as its own standalone component?

rkzel replied:
It was one cleanup I wasn't convinced needed to be done immediately - but you're right, I've split it into 3 components.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -113,110 +121,187 @@ const NewProject = () => {
+GitHubRepoList.propTypes = PropTypes.func.isRequired
+
+const createProject = () => {
+  console.log('creating native project')
```
Ah, it looks like this PR is meant to be left unmerged until we implement the rest of this? 

Can we build this in a "feature flipped" way, so that we don't have to delay merging this PR for an extended period of time, but we also don't have unsupported elements in our UI? Ideally, we could do this with an environment variable, or maybe a URL param (like, add `?features=nativeprojects` to the end of the URL), but for an initial version I think we could even have something like `if (false)` in the code somewhere, to make it easy to toggle on & off in a development environment.

rkzel replied:
It was meant to be merged at more or less the same time as the rest of it - but "feature flipped" makes sense as safety measure, to make sure there are no dangling threads accidentaly deployed where they shouldn't.
The easiest way in this particular case was to disable the -OR- part with a small `true?` statement - not very elegant, but it felt justified...

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -113,110 +121,187 @@ const NewProject = () => {
+      {title !== '' ? (
+        <>
+        <FormField
+          label="Feedback"
```
Wrong field name

rkzel replied:
Amazing how so glaringly obvious can be missed. Fixed.

chadoh reviewed (changes requested):
Code looks pretty clean, but there are some changes I'd like to see, in addition to the UI changes @stellarmagnet requested above

rkzel committed:
refactor: github query component and post-review fixes

rkzel commented:
> * The padding is not correct between the field labels. Please double check all of the padding between elements.

Improved that by fixing height line and height of the field label component - which is used everywhere in the app, so I made it as much as I could without checking all the other places.

> * For the "And when GH is disconnected" use case --> the blue box is too close to the field label

Fixed. Styled-components module problem.

> * "Feedback" should say "Description"

Sorry, fixed.

> * Select from Github icon & text should be center-aligned with the button

Improved!

Screenshots added to the first, top comment.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -1,41 +1,87 @@
   const [ filter, setFilter ] = useState('')
-  const [ project, setProject ] = useState()
-  const [ repoSelected, setRepoSelected ] = useState(-1)
+  const searchRef = useRef(null)
```
Small note: I don't think you need the `null` here. `useRef()` will initialize just fine without it

rkzel replied:
I think so - fixed.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +204,127 @@ const NewProject = () => {
+
+      {
+        // eslint-disable-next-line no-constant-condition
+        true ? (
```
This isn't the part we need to feature flip!

Until native projects are actually supported, we want to hide everything related to native projects. We want the experience to match what's in mainnet & rinkeby today. 

That means we _only_ want the `GitHubRepoList`. We want to hide the "Title" field. We want to hide the `InfoBox`.

chadoh replied:
Ah, I see – only the InfoBox was missed. The indentation of the part below had me thinking this was more broken than it is

rkzel replied:
Added InfoBox to the excluded part and fixed misleading intendation.

chadoh commented:
@rkzel I explored feature-flipping using URL query params but that [won't work](https://github.com/AutarkLabs/open-enterprise/pull/1918). How about instead we only show the Native Projects stuff if the `connectedAccount` matches an array of hard-coded addresses? We can add our development superuser to this list (`0xb412`) as well as the Rinkeby addresses of @stellarmagnet and @javieralaves.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +204,127 @@ const NewProject = () => {
+          <GitHubRepoList handleGithubSignIn={handleGithubSignIn} />
+        ) : (
+        <>
+      <FormField
```
indentation seems way off

rkzel replied:
Yes. Fixed it in case it's going to be needed for more than one day.

rkzel committed:
fix: hidden InfoBox, adjusted ScrollList height, fixed intendation

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,128 @@ const NewProject = () => {
+  return (
+    <PanelContent>
+      {
+        // eslint-disable-next-line no-constant-condition
```
you don't need this comment anymore

rkzel replied:
True. Removed.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
+    </>
+  )
+}
+GitHubRepoList.propTypes = PropTypes.func.isRequired
```
what's this?

rkzel replied:
It's a shortcut. In case there is only one property one does not need to write:

```
GitHubRepoList.propTypes = {
   handleGithubSignIn: PropTypes.func.isRequired
}
```

it is sufficient to say:

`GitHubRepoList.propTypes = PropTypes.func.isRequired`

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
+                color={`${theme.surfaceContentSecondary}`}
+                linecolor={`${theme.border}`}
+              >
+                  OR
```
indentation is off

rkzel replied:
And that got fixed as well...

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
+              <>
+              <ThematicBreak
+                color={`${theme.surfaceContentSecondary}`}
+                linecolor={`${theme.border}`}
```
I think it would clean up this component nicely if these props didn't have to be passed in

```jsx
const ThematicBreak = ({ children }) => {
  const theme = useTheme()
  return (
    <div css={`
      ...
    `}>
      {children}
    </div>
  )
}
```

Or maybe if the "or" is also included in the component itself:

```jsx
const ThematicBreak = () => {
  const theme = useTheme()
  return (
    <div css={`
      ...
    `}>
      or
    </div>
  )
}
```

rkzel replied:
Ok. The scecond one makes sense - this component is only used once, it will probably retain colors and text if it needs to be reused, and it can alway be re-customized later.. Changed.

chadoh commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
+      {
+        disableDecoupledProjects ? (
+          <GitHubRepoList handleGithubSignIn={handleGithubSignIn} />
+        ) : (
```
an early return would look nicer, keep the indentation the way you had it, and make it easier to remove this feature-flipping later:

```jsx
if (disableDecoupledProjects) {
  return (
    <PanelContent>
      <GitHubRepoList handleGithubSignIn={handleGithubSignIn} />
    </PanelContent>
  )
}

return (
  <PanelContent>
    ...
  </PanelContent>
)
```

rkzel replied:
Good point. Removing feature-flipping later will produce much less commit noise too. Changed.

chadoh reviewed (dismissed):
Please at least fix the invalid propTypes before merging, if not also my other comments

Otherwise, it seems this works as desired

rkzel committed:
refactor: feature-flip, ThematicBreak

Quazia commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,151 @@ const NewProject = () => {
+  const [ description, setDescription ] = useState('')
+  const { appState: { github = { status: STATUS.INITIAL } } } = useAragonApi()
+
+  if (disableDecoupledProjects) return (
```
This needs to be removed.

Quazia commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -1,41 +1,89 @@
+import { FormField, FieldTitle } from '../../Form'
+import { STATUS } from '../../../utils/github'
 
+const disableDecoupledProjects = true
```
Get rid of this too.

Quazia commented on apps/projects/app/components/Panel/NewProject/NewProject.js:
```diff
@@ -63,160 +206,151 @@ const NewProject = () => {
+  )
+}
+
+const createProject = () => {}
```
Add the logic in here.

Quazia reviewed (approved):
Added some comments for once we have the contract changes merged but it's looking good to merge.
